### DISCUSSION TITLE
Weston or eglfs?

### DISCUSSION THREAD

**Discussion by gencay3535** [meta-flutter/meta-flutter#172] [Q&A] [answered]
Hello, which one do you think is better for using flutter with st boards weston or eglfs?

**Comment by jwinarske** [comment]
You might want to run the `sdcard_stm32mp157f-dk2-optee-weston-runtimerelease` image on disco board: 
https://github.com/meta-flutter/meta-flutter/actions/runs/2952690852

My observation is that flutter-auto runs smoother than flutter-pi does on STM32MP15.  flutter-pi has visual anomalies, as well as the cursor needs a special branch.

To disable Weston on the STM32MP15 Weston image
```
systemctl disable weston
reboot
```

To re-enable Weston on the STM32MP15 Weston image
```
systemctl enable weston
reboot
```

**Reply by gencay3535** [reply]
I was looking at the yaml file and you have two builds one was for weston and other one was for eglfs, what is the difference between them?

**Reply by jwinarske** [reply]
One is for DRM, one is for Weston.  Beyond this the primary difference is the OS image size.  One could remove more items on the Weston image and get it smaller.  The images tend to include the kitchen sink.

**Reply by jwinarske** [reply, marked as answer]
eglfs doesn't include a compositor which would be for DRM only use cases (flutter-pi). The Weston image works for either flutter-pi or flutter-auto.  You need to disable the Weston service for DRM use cases (original response has steps).  flutter-auto has the most features running on the AGL compositor (Multi-view per compositor pane).
